pilot/pkg/config/coredatamodel/controller.go

```diff
@@ -165,29 +169,57 @@ func (c *Controller) Apply(change *sink.Change) error {
 			return err
 		}
 
-		namedConfig, ok := innerStore[conf.Namespace]
-		if ok {
-			namedConfig[conf.Name] = conf
-		} else {
-			innerStore[conf.Namespace] = map[string]*model.Config{
-				conf.Name: conf,
+		switch {
+		case change.Incremental:
+			// Incremental EDS is only available for SyntheticServiceEntries currently.
+			// EDS updates only take place if annotations for that SyntheticServiceEntry
+			// is only set to endpointVersion and NOT serviceVersion or both
+			if descriptor.Collection == model.SyntheticServiceEntry.Collection &&
+				updateEndpoints(conf.Annotations) {
+				var err error
+				// send eds Update
+				if edsUpdated, err = c.edsUpdate(conf); err != nil {
+					return err
+				}
+			}
+			// update configStore incrementally
+			c.incrementalConfigStore(descriptor.Type, conf, change.Removed)
```

Comment:
if only endpoints changed, do we still need to do this?

Reply:
This was also one of my questions in the meeting, “do we update the `configStore` when we issue an edsUdpate?”  if I remember correctly @costinm  mentioned we still want to update the configStore. @costinm  is that still the case?

Reply:
Yes, for now it is still needed. We may remove it after Consul is moved out, the List is only called to support 
consul right now ( because it's not updating the endpoint shards ). We also need to check.